src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/AES/AesCornerTests.cs

```diff
@@ -324,5 +324,115 @@ public static void ZeroLengthTransformPKCS7()
+
+            using (Aes a = Aes.Create())
+            {
+                a.Key = key;
+                a.IV = iv;
+                a.Mode = CipherMode.CBC;
+                a.Padding = paddingMode;
+
+                byte[] cipher = a.Encrypt(plainText);
+
+                // we cannot validate the cipher in this padding mode as it consists of random data
+                if (paddingMode != PaddingMode.ISO10126)
+                {
+                    Assert.Equal<byte>(expectedCipher, cipher);
+                }
+
+                // decrypt it with PaddingMode.None so that we can inspect the padding manually
+                a.Padding = PaddingMode.None;
+                byte[] decrypted = a.Decrypt(cipher);
+                ValidatePadding(decrypted, paddingMode, expectedPaddingSize);
+            }
+        }
+
+        private static void ValidatePadding(byte[] buffer, PaddingMode paddingMode, int expectedPaddingSize)
+        {
+            switch (paddingMode)
+            {
+                case PaddingMode.PKCS7:
+                    ValidatePKCS7Padding(buffer, expectedPaddingSize);
+                    break;
+                case PaddingMode.ANSIX923:
+                    ValidateANSIX923Padding(buffer, expectedPaddingSize);
+                    break;
+                case PaddingMode.ISO10126:
+                    ValidateISO10126Padding(buffer, expectedPaddingSize);
+                    break;
+                case PaddingMode.Zeros:
+                    ValidateZerosPadding(buffer, expectedPaddingSize);
+                    break;
+                case PaddingMode.None:
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        private static void ValidateZerosPadding(byte[] buffer, int expectedPaddingSize)
+        {
+            for (int i = buffer.Length - 1; i > buffer.Length - 1 - expectedPaddingSize; i--)
+            {
+                Assert.Equal(0, buffer[i]);
+            }
+        }
+
+        private static void ValidatePKCS7Padding(byte[] buffer, int expectedPaddingSize)
+        {
+            for (int i = buffer.Length - 1; i > buffer.Length - 1 - expectedPaddingSize; i--)
+            {
+                Assert.Equal(expectedPaddingSize, buffer[i]);
+            }
+        }
+
+        private static void ValidateANSIX923Padding(byte[] buffer, int expectedPaddingSize)
+        {
+            Assert.Equal(buffer[buffer.Length - 1], expectedPaddingSize);
+
+            for (int i = buffer.Length - expectedPaddingSize; i < buffer.Length - 1; i++)
+            {
+                Assert.Equal(0, buffer[i]);
+            }
+        }
+
+        private static void ValidateISO10126Padding(byte[] buffer, int expectedPaddingSize)
+        {
+            // there is nothing else to validate as all the other padding bytes are random.
+            Assert.Equal(buffer[buffer.Length - 1], expectedPaddingSize);
```

Comment:
So you are saying randomly pick one of the bytes and check if it is non-zero? 
I can add it, but I don't think it will add too much value. I submitted #13902 -- we can chat more there.